netclient/functions/join.go

```diff
@@ -238,6 +238,14 @@ func JoinNetwork(cfg *config.ClientConfig, privateKey string) error {
 			logger.Log(1, "network:", cfg.Network, "error retrieving private address: ", err.Error())
 		}
 	}
+	if len(cfg.Node.Interfaces) == 0 {
+		ip, err := getInterfaces()
+		if err != nil {
+			logger.Log(0, "failed to retrive local interfaces", err.Error())
```

Comment:
if there is an error, should we set interfaces to an empty array, in-case it's nil?

Reply:
same to below file

Reply:
the only way this interfaces can get set to nil is if getInterfaces() returns nil,nil , ie no errors retrieving interfaces but no actual interfaces.  If someone is running netclient with no interfaces a panic isn't out of the question.

if the case of an error from getInterfaces(), continuing to use the previously generated array is less misleading than setting interfaces to an empty array.